client/app/services/dialog-field-refresh.service.js

```diff
@@ -24,6 +24,12 @@
           }
         });
 
+        _.find(allDialogFields, function(dialogField) {
```

Comment:
Wait, this is wrong.. `find` is supposed to return the first item for which the callback returns true..

You sure you didn't mean something like.. ?

```js
dialogFieldsToRefresh.forEach(function(name) {
  _.find(allDialogFields, { name: name }).beingRefreshed = true;
});
```